src/subscription/subscribe.js

```diff
@@ -160,7 +159,7 @@ function subscribeImpl(
     // Note: Flow can't refine isAsyncIterable, so explicit casts are used.
     isAsyncIterable(resultOrStream)
       ? mapAsyncIterator(
-          ((resultOrStream: any): AsyncIterable<mixed>),
+          resultOrStream,
```

Comment:
@IvanGoncharov since adding `boolean %checks(value instanceof AsyncIterable)`  to `isAsyncIterable` I was able to remove the cast here, but I was not able to remove it from the other side of the ternary.

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/subscription/subscribe.js:166:9

Cannot call sourcePromise.then with function bound to onFulfill because mixed [1] is incompatible
with object type [2] in type argument Yield [3] of the return value of property @@asyncIterator of
the return value. [incompatible-call]

     src/subscription/subscribe.js
 [2] 117│ ): Promise<AsyncIterator<ExecutionResult> | ExecutionResult> {
        :
     163│           mapSourceToResponse,
     164│           reportGraphQLError,
     165│         )
     166│       : resultOrStream,
     167│   );
     168│ }
     169│
        :
 [1] 206│ ): Promise<AsyncIterable<mixed> | ExecutionResult> {

     /private/tmp/flow/flowlib_5bfb8b1/core.js
 [3] 562│ interface $AsyncIterator<+Yield,+Return,-Next> {
```